internal/it/test_skip.go

```diff
@@ -0,0 +1,221 @@
+package it
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/hazelcast/hazelcast-go-client/internal"
+)
+
+const (
+	skipHzVersion     = "hz"
+	skipClientVersion = "ver"
+	skipOS            = "os"
+	skipEnterprise    = "enterprise"
+	skipOSS           = "oss"
+	enterpriseKey     = "HAZELCAST_ENTERPRISE_KEY"
+)
+
+// SkipIf can be used to skip a test case based on comma-separated conditions.
+// The following conditions may be set:
+// "hz": Hazelcast version
+// "ver": go client version
+// "os": value of runtime.GOOS
+// "enterprise"/"oss": presence of enterprise key environment variable
+// Example: SkipIf(t, "ver > 1.1, hz = 5")
+func SkipIf(t *testing.T, conditionString string) {
+	skip, err := skipIf(conditionString)
+	if err != nil {
+		t.Fatalf(err.Error())
+	}
+	if skip {
+		t.Skip("Skipping test: SkipIf conditions met")
+	}
+}
+
+func skipIf(conditionString string) (bool, error) {
+	conditions := strings.Split(conditionString, ",")
+	for _, condition := range conditions {
+		condition = strings.Trim(condition, " ")
+		switch parts := strings.Split(condition, " "); {
+		case parts[0] == skipHzVersion:
+			if err := validateLength(parts, 3, condition, "hz = 4.0"); err != nil {
+				return true, err
+			}
+			skip, err := checkVersion(parts[2], parts[1], HzVersion())
+			if err != nil {
+				return true, err
+			}
+			if skip {
+				return true, nil
+			}
+		case parts[0] == skipClientVersion:
+			if err := validateLength(parts, 3, condition, "ver = 4.0"); err != nil {
+				return true, err
+			}
+			skip, err := checkVersion(parts[2], parts[1], internal.ClientVersion)
+			if err != nil {
+				return true, err
+			}
+			if skip {
+				return true, nil
+			}
```

Comment:
It seems we repeat the code a bit here. Would it make sense to remove code duplication?

Reply:
I've cleaned up this part a bit. I think it is a lot simpler now.